plugin/src/main/java/org/gradle/testretry/internal/executer/RetryTestResultProcessor.java

```diff
@@ -90,7 +94,11 @@ public void completed(Object testId, TestCompleteEvent testCompleteEvent) {
 
                 boolean failedInPreviousRound = previousRoundFailedTests.remove(className, name);
                 if (failedInPreviousRound && testCompleteEvent.getResultType() == SKIPPED) {
-                    currentRoundFailedTests.add(className, name);
+                    if (classRetryMatcher.retryWholeClass(className)) {
```

Comment:
:thinking Wrap this part in private method so it can be reused from below as well?